src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/IncentiveLevelService.kt

```diff
@@ -73,15 +85,36 @@ class IncentiveLevelService(
   @Transactional
   suspend fun updateIncentiveLevel(code: String, update: IncentiveLevelUpdateDTO): IncentiveLevelDTO? {
     return incentiveLevelRepository.findById(code)
-      ?.let {
-        incentiveLevelRepository.save(it.withUpdate(update))
+      ?.let { originalIncentiveLevel ->
+        val incentiveLevel = originalIncentiveLevel.withUpdate(update)
+
+        if (!incentiveLevel.active && incentiveLevel.required) {
+          throw ValidationException("A level must be active if it is required")
+        }
+
+        incentiveLevelRepository.save(incentiveLevel)
           .toDTO()
           .also {
+            if (it.required && !originalIncentiveLevel.required) {
+              // This probably could be moved out of the transaction once auditing is moved out:
+              enablePrisonIncentiveLevelEverywhere(it.code)
+            }
             auditService.sendMessage(INCENTIVE_LEVEL_UPDATED, it.code, it)
           }
       }
   }
 
+  // TODO: Many DB queries are made: does this warrant NOT being in a transaction?
```

Comment:
Won't this be in a transaction anyway as its private and I assume `@Transactional` is already setup higher up

Reply:
the methods that call this are transactional currently, my todo comments suggests maybe moving this out of the transaction. here's some more discussion on that point:

https://github.com/ministryofjustice/hmpps-incentives-api/pull/391#discussion_r1149167238